### ISSUE TITLE
The source anime planet doesn't have the tags categorised

### ISSUE THREAD

**Issue by MrAndre96** [Bionus/imgbrd-grabber#1383]
**Bug description**

When you use the source anime planet and click on a image you don't see the different kinds of copyright like with other sources. (For example purple for copyright and green for characters)
But like you can see here they do have that on their website: [https://i.imgur.com/nl7MO9A.png](https://i.imgur.com/nl7MO9A.png)

**Steps to reproduce**

Use the source anime planet and search for nekopara for example
Now click on a image and you will see that none of the tags are categorised like by the other sources

**Expected behavior**

Have them categorised with the different colors

**System information**

Windows 10 with the latest nightly version

**Comment by Bionus** [comment]
Indeed!
Using the JSON API for loading image tags (as found in issue #1051) seems to fix the issue. :+1:

**Comment by MrAndre96** [comment]
@Bionus  There is a new bug caused by this patch I think that make the html source load 2 times behind each other and with the second time reducing the filesize to 88,5 kb as you can see here: [https://i.imgur.com/X0NpUIN.png](https://i.imgur.com/X0NpUIN.png)
Also the tags haven't been categorised with both the json or html source enabled as you can see here: [https://i.imgur.com/PcBxcap.png](https://i.imgur.com/PcBxcap.png). Would like to see them like this: [https://i.imgur.com/qeg8N82.png](https://i.imgur.com/qeg8N82.png) which should I think be possible since they are categorised on their website

**Comment by Bionus** [comment]
> @Bionus There is a new bug caused by this patch I think that make the html source load 2 times behind each other and with the second time reducing the filesize to 88,5 kb as you can see here: https://i.imgur.com/X0NpUIN.png

I can't reproduce that with that image ([ID 469261](https://anime-pictures.net/pictures/view_post/469261?lang=en&type=json) as far as I can see), the image filesize is always shown as 1.12 MB.

Can I see your log? (from the moment the image window is open is enough) 👍 

> Also the tags haven't been categorised with both the json or html source enabled as you can see here: https://i.imgur.com/PcBxcap.png. Would like to see them like this: https://i.imgur.com/qeg8N82.png which should I think be possible since they are categorised on their website

For the moment the tags are only categorized with the JSON source. Doing it with the HTML one is technically possible but would be very easy to break as the way they are shown on the website is not explicit, using colors instead of types (so grabber would need to map the colors to their corresponding types, which would stop working as soon as the color changes).

I might implement it in the future, but given the JSON API works very well, I can only advice to use it instead.

**Comment by MrAndre96** [comment]
`[18:54:01.858][Info] [anime-pictures.net][Html] Loading page https://anime-pictures.net/pictures/view_posts/0?search_tag=nekopara&lang=en
[18:54:02.001][Info] [anime-pictures.net][Html] Receiving page https://anime-pictures.net/pictures/view_posts/0?search_tag=nekopara&lang=en
[18:54:02.022][Info] [anime-pictures.net][Html] Parsed page https://anime-pictures.net/pictures/view_posts/0?search_tag=nekopara&lang=en: 60 images (60), 0 tags (0), -1 total (160), 2 pages (2)
[18:54:02.224][Warning] [Qt][default] libpng warning: iCCP: known incorrect sRGB profile
[18:54:05.969][Info] Loading image from https://anime-pictures.net/pictures/download_image/541376.jpg
[18:54:05.981][Info] Loading and saving image in C:\Users\andre\AppData\Local\Temp\Grabber\f42ef58e-0de6-473c-8139-d38bcca9cfcf.jpg
[18:54:06.027][Info] Loading and saving image in C:\Users\andre\AppData\Local\Temp\Grabber\f42ef58e-0de6-473c-8139-d38bcca9cfcf.jpg
[18:54:06.138][Info] Loading and saving image in C:\Users\andre\AppData\Local\Temp\Grabber\f42ef58e-0de6-473c-8139-d38bcca9cfcf.jpg
[18:54:06.423][Info] Image received from https://anime-pictures.net/pictures/download_image/541376.jpg
[18:55:34.972][Warning] [Qt][default] libpng warning: iCCP: known incorrect sRGB profile
[18:55:47.828][Info] Saving sources...
[18:55:47.828][Info] [anime-pictures.net] Logging in...
[18:55:47.829][Info] Updating checkboxes.
[18:55:48.265][Info] [anime-pictures.net] Login finished: success.
[18:55:49.449][Info] Loading results...
[18:55:49.450][Info] [anime-pictures.net][Html] Loading page https://anime-pictures.net/pictures/view_posts/0?search_tag=nekopara&lang=en
[18:55:49.451][Info] [anime-pictures.net][Json] Loading page https://anime-pictures.net/pictures/view_posts/0?search_tag=nekopara&posts_per_page=20&lang=en&type=json
[18:55:49.722][Info] [anime-pictures.net][Html] Receiving page https://anime-pictures.net/pictures/view_posts/0?search_tag=nekopara&lang=en
[18:55:49.742][Info] [anime-pictures.net][Html] Parsed page https://anime-pictures.net/pictures/view_posts/0?search_tag=nekopara&lang=en: 60 images (60), 0 tags (0), -1 total (160), 2 pages (2)
[18:55:49.753][Info] [anime-pictures.net][Json] Receiving page https://anime-pictures.net/pictures/view_posts/0?search_tag=nekopara&posts_per_page=20&lang=en&type=json
[18:55:49.761][Info] [anime-pictures.net][Json] Parsed page https://anime-pictures.net/pictures/view_posts/0?search_tag=nekopara&posts_per_page=20&lang=en&type=json: 20 images (20), 0 tags (0), 124 total (124), 7 pages (7)
[18:56:02.497][Info] Loading image from https://anime-pictures.net/pictures/get_image/541376-1580x1020-neko paradise-neko works (studio)-coconut (sayori)-sayori-long hair-single.png
[18:56:02.509][Info] Loading and saving image in C:\Users\andre\AppData\Local\Temp\Grabber\378f4047-f0f2-4d27-bbcd-2a3f245091f8.png
[18:56:02.945][Info] Image received from https://anime-pictures.net/pictures/get_image/541376-1580x1020-neko paradise-neko works (studio)-coconut (sayori)-sayori-long hair-single.png`
As you can see it is loading a image from https://anime-pictures.net/pictures/download_image/541376.jpg which isn't a image by the html source

**Comment by MrAndre96** [comment]
@Bionus The tags are only categorised inside the more details window here: [https://i.imgur.com/mxLBWiE.png](https://i.imgur.com/mxLBWiE.png) but not here inside the image viewer: [https://i.imgur.com/n84AoDq.png](https://i.imgur.com/n84AoDq.png)

**Comment by MrAndre96** [comment]
It is now by both in alphabetic order and not categorised (with copyright first etc): [https://i.imgur.com/RISBQEV.png](https://i.imgur.com/RISBQEV.png) using the json source

**Comment by Bionus** [comment]
Yeah, I went to sleep and didn't have time to finish this yesterday. 😄 

You're very likely missing the `tag-types.txt` file in your `C:/Users/Nicolas/AppData/Local/Bionus/Grabber/sites/Anime pictures/anime-pictures.net/` folder, which wasn't included in the setup (it should be now).
